lib/error.js

```diff
@@ -41,6 +41,25 @@ ShareDBError.CODES = {
   ERR_OT_OP_NOT_PROVIDED: 'ERR_OT_OP_NOT_PROVIDED',
   ERR_PROTOCOL_VERSION_NOT_SUPPORTED: 'ERR_PROTOCOL_VERSION_NOT_SUPPORTED',
   ERR_QUERY_EMITTER_LISTENER_NOT_ASSIGNED: 'ERR_QUERY_EMITTER_LISTENER_NOT_ASSIGNED',
+  /**
+   * A special error that a "readSnapshots" middleware implementation can use to indicate that it
+   * wishes for the ShareDB client to treat it as a silent rejection, not passing the error back to
+   * user code.
+   *
+   * For subscribes, the ShareDB client will still cancel the document subscription.
+   */
+  ERR_SNAPSHOT_READ_SILENT_REJECTION: 'ERR_SNAPSHOT_READ_SILENT_REJECTION',
+  /**
+   * A "readSnapshots" middleware rejected the reads of specific snapshots.
+   *
+   * This error code is mostly for server use and generally will not be encountered on the client.
+   * Instead, each specific doc that encountered an error will receive its specific error.
+   *
+   * The one exception is for queries, where a "readSnapshots" rejection of specific snapshots will
+   * cause the client to receive this error for the whole query, since queries don't support
+   * doc-specific errors.
+   */
+  ERR_SNAPSHOT_READS_REJECTED: 'ERR_SNAPSHOT_READS_REJECTED',
```

Comment:
I'd like to get rid of this error and only have the individual errors on the snapshots themselves

Reply:
This error is still needed, as it's sent to the client for queries, when at least one snapshot in the query results gets rejected, as the comment indicates.